### ISSUE TITLE
Unchangable hard-coded variables

### ISSUE THREAD

**Issue by erik-ekberg-sg** [jesusdalvarado/rapids-v1#1]
When we build packages it is okay to tailor them to our specific use case (i.e. pre-installed Ably, Cloudevents, etc.) but we should be mindful to not couple them too tightly to how we are doing things at this very moment in time as the package should be somewhat flexible.

Abstracting tasks like "subscribe to some Ably channel" or "create some cloudevent" are good tasks to abstract into this package to reduce boilerplate. But locking users into specific configuration details is not recommended. Hard-coded values like `ABLY_DEVELOPMENT_CHANNEL_NAME=abcdefg` are an implementation detail for users, like our future selves, to decide on themselves. Another example is `rapids-v1-hydrator-v0`. Hard-coding these values requires an API-breaking change for something that ultimately is a concern of the user but not of the package. An example of something that should be a concern to the package is swapping providers from Ably to Pusher for example, or changing the interface of the "create some cloudevent" function. What cloudevents users create or what channels users subscribe to is not so much a concern of the package.

I recommend making this configurable either through environment variables or preferably through environmental variables and in-code configuration variables

```node
const { Rapids } = require('...')
const rapids = new Rapids({ channel: process.env.JESUS_RAPIDS_ABLY_CHANNEL_NAME || 'my-unique-channel-name' })
```

**Comment by jesusdalvarado** [comment]
@erik-ekberg-sg Initially I just implemented this package for development, but now that we are thinking of using it also in frontend (which I totally agree we should) I was thinking of detecting the environment within the package and automatically decide between the dev_ably_channel vs the prod_ably_channel. This is because we have been using only these 2 environments during the current year, so it makes sense to adapt it to our use case so we don't have to set it every single time we use this package. Although, I agree giving the option to set it as an environmental variable can be useful, if for whatever reason we decided to implement it in a different environment in the future.

**Comment by erik-ekberg-sg** [comment]
@jesusdalvarado, agreed, I would do it similar to the `@1mill/sops` implementation where you can override the `file` argument but it defaults to `prod.secrets.sops.json` or `dev.secrets.sops.json` if that override is not present - https://github.com/1Mill/sops/blob/436608158c8380d37132120f1afd5c964580035c/src/index.js#L10

The only pitfall of defaults is that if they change it usually means an API breaking change but that is likely worth it in this case for the ease of use.
